jaxrs-clients/src/test/java/com/palantir/remoting3/jaxrs/feignimpl/Java8TestServer.java

```diff
@@ -46,6 +47,7 @@
     public final void run(Configuration config, final Environment env) throws Exception {
         env.jersey().register(HttpRemotingJerseyFeature.INSTANCE);
         env.jersey().register(new TestResource());
+        env.jersey().register(new EmptyOptionalTo204ExceptionMapper());
```

Comment:
why do we need this?

Reply:
Dropwizard 1.x converts Optional.empty to 404 which breaks our tests. While 0.x version did nothing special about Optionals we need this now to preserve the expected behaviour of returning 204

Reply:
https://github.com/dropwizard/dropwizard/pull/2350 added this class in dropwizard

Reply:
we'll need to minimally understand (and maybe deduplicate) this with Java8OptionalMessageBodyWriter and NoContentExceptionMapper

Reply:
(because it's not obvious how to use a HttpRemotingJerseyFeature.INSTANCE with dropwizard to get the desired behavior)

Reply:
but then I guess it's called JerseyFeature, not DropwizardFeature. not your fault, let's move on. https://github.com/palantir/http-remoting/issues/714